doc/nrf/ug_thread_architectures.rst

```diff
@@ -0,0 +1,162 @@
+.. _ug_thread_architectures:
+
+OpenThread architectures
+########################
+
+This page describes the OpenThread stack architecture and platform designs that are possible with the OpenThread network stack on Nordic Semiconductor devices in |NCS|.
+
+The designs are described from the least to the most complex, that is from simple applications that consist of a single chip running single or multiple protocols to scenarios in which the nRF SoC acts as a network co-processor when the application is running on a much more powerful host processor.
+
+.. contents::
+    :local:
+    :depth: 1
+
+The platform designs work with the following development kits:
+
+.. include:: /includes/boardname_tables/sample_boardnames.txt
```

Comment:
As described in JIRA, we would like rather list possible SoC for specific designs than show the entire list here. These lists should be placed under a specific design description.

For example:
List of suitable SoC for:
SoC: nRF52840, nRF52833, nRF5340 (not yet supported)
Mutli: nRF52840, nRF52833, nRF5340 (not yet supported)
NCP: nRF52840, nRF52833, nRF5340 (not yet supported)
RCP:  nRF52811 (not yet supported), nRF52820 (not yet supported)